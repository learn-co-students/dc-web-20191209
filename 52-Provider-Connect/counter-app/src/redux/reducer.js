const reducer = (oldState = {count: 123}, action) => {
  switch(action.type){
    case "INCREMENT":
      return {count: oldState.count + action.payload}
    case "DECREMENT":
      return {count: oldState.count - action.payload}
    default:
      return oldState
  }
}

export default reducer
