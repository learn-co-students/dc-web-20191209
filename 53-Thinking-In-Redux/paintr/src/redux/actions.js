
function vote(paintingId){
  return {type:"INCREASE_VOTE", paintingId}
}

function search(newSearchTerm){
  return {type: "CHANGE_SEARCH_TEXT", payload: newSearchTerm}
}

export {vote, search}
//search() //{type: "CHANGE_SEARCH_TEXT", payload: newSearchTerm}
