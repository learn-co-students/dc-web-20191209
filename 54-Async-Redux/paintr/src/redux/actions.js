// action creators
const URL = "http://localhost:3000/paintings"

function changeSearchText(value) {
  return { type: "CHANGE_SEARCH_TEXT", payload: value };
}

function votedForPainting(paintingId) {
  return { type: "INCREASE_VOTES", payload: paintingId };
}

function votingForPainting(paintingId) {
  return (dispatch, getState) => {
    let oldVotes = getState().paintings.find(p => p.id === paintingId).votes
    fetch(`${URL}/${paintingId}`, {
      method: "PATCH",
      headers: {
        "Content-Type": "application/json",
        "Accept": "application/json"
      },
      body: JSON.stringify({votes: oldVotes + 1})
    })
  }
}

function updatePainting({ title, name, birthday, deathday, paintingId }) {
  return {
    type: "UPDATE_PAINTING",
    payload: { title, name, birthday, deathday, paintingId}
  };
}

function fetchedPaintings(paintings) {
  return {type: "FETCHED_PAINTINGS", payload: paintings}
}

function fetchingPaintings() {
  return (dispatch) => {
    fetch(URL)
    .then(res => res.json())
    .then(paintings => {
      dispatch(fetchedPaintings(paintings))
    })
  }
}

export { changeSearchText, votingForPainting, updatePainting, fetchingPaintings };
