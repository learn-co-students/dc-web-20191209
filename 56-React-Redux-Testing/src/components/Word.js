import React, { Component } from 'react';
import { connect } from 'react-redux';

class Word extends Component {
    render() {
        return (
            <div className='word-card'>
                <h2>{this.props.word.name}</h2>
                <img className="word-image" src={this.props.word.img_url} /><br></br>
                <button onClick={() => this.props.deleteWord(this.props.word.id)} >Delete</button>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        deleteWord: (wordId) => dispatch({ type: 'DELETE_WORD', payload: wordId})
    }
}

export default connect(null, mapDispatchToProps)(Word)