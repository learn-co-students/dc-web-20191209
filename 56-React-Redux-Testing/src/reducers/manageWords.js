let idTracker = 1;

export default function manageWord(state = { words: [] }, action) {
  switch (action.type) {
    case "ADD_WORD":
      let word = {
        id: idTracker,
        img_url: action.payload.img_url,
        name: action.payload.name
      };
      idTracker++;
      console.log(word);
      return { ...state, words: [...state.words, word] };
    case "DELETE_WORD":
      let filteredWords = state.words.filter(
        word => word.id !== action.payload
      );
      return { ...state, words: filteredWords };
    default:
      return state;
  }
}
