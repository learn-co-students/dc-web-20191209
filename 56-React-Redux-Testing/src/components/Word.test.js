import React from "react";
import Word from "./Word";
import { createStore } from "redux";
import manageWords from "../reducers/manageWords";
import renderer from "react-test-renderer";

const mockStore = createStore(manageWords);

let mockProps = {
  word: { id: 2, name: "vaca", img_url: null, props: jest.fn() }
};

describe("Word", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Word {...mockProps} store={mockStore} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
