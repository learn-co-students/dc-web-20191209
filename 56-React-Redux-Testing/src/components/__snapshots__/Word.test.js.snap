// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Word renders correctly 1`] = `
<div
  className="word-card"
>
  <h2>
    vaca
  </h2>
  <img
    className="word-image"
    src={null}
  />
  <br />
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
